pkg/ec/error.go

```diff
@@ -137,6 +138,22 @@ func NewErrNotImplemented() *ControllerError {
 	return NewControllerError(http.StatusNotImplemented)
 }
 
+// IsRetryable returns true and the retry delay if ANY errors eminating from the supplied error
+// is an ec.ControllerError that is Retryable, and false otherwise.
+func IsRetryable(err error) (bool, time.Duration) {
+
+	for err != nil {
+		var ctrlErr *ControllerError
+		if errors.As(err, &ctrlErr) && ctrlErr.IsRetryable() {
+			return true, ctrlErr.RetryDelay()
+		}
+
+		err = errors.Unwrap(err)
```

Comment:
Can this loop get stuck?

Reply:
Maybe? If you f'd up and your Unwrap() returns itself